spec/support/shared_examples_policy_machine.rb

```diff
@@ -820,4 +820,85 @@
 
   end
 
+
+  describe 'batch_find' do
+
+    before do
+      @one_fish = policy_machine.create_object('one:fish')
+      @two_fish = policy_machine.create_object('two:fish')
+      @red_one = policy_machine.create_object('red:one')
+      @read = policy_machine.create_operation('read')
+      @write = policy_machine.create_operation('write')
+      @u1 = policy_machine.create_user('u1')
+      @ua = policy_machine.create_user_attribute('ua')
+      [@one_fish, @two_fish, @red_one].each do |object|
+        policy_machine.add_association(@ua, Set.new([@read]), object)
+      end
+      @oa = policy_machine.create_object_attribute('oa')
+      policy_machine.add_association(@ua, Set.new([@write]), @oa)
+      policy_machine.add_assignment(@u1, @ua)
+      policy_machine.add_assignment(@red_one, @oa)
+    end
+
+    context 'when given a block' do
```

Comment:
This should also have a test that [asserts that the block is called](https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/yield-matchers). Otherwise since all your assertions are inside the passed-in block, a method that didn't do anything would still pass all the specs.